Sources/SwiftProcessing/Core/Sketch3D/Sketch3DTransform.swift

```diff
@@ -2,103 +2,102 @@ import UIKit
 import SceneKit
 
 public extension Sketch {
-    
-    func translationNode(_ vector: SCNVector3,_ property: String) {
-        
+
+    func translationNode(_ vector: SCNVector3,_ property: String, _ checkNoShapeNodes: Bool = true, _ changeCurrentTransitionNode: Bool = true) {
         let lastNode = currentTransformationNode
-        
-        if lastNode.hasNoShapeNodes() {
-            
+
+        if lastNode.hasNoShapeNodes() && checkNoShapeNodes {
             switch property {
-                
+
             case "position":
                 lastNode.position.add(vector)
-                
+
             case "rotation":
                 lastNode.eulerAngles.add(vector)
-                
+
             default:
                 print("Wrong translation property key word")
-                
+
             }
-                        
+
         } else if lastNode.hasAvailableTransitionNodes() {
-            
             let nextNode = lastNode.getNextAvailableTransitionNode()
-            
+
             switch property {
-                
+
             case "position":
                 nextNode.position = vector
                 nextNode.eulerAngles = SCNVector3(0,0,0)
-                
+
             case "rotation":
                 nextNode.eulerAngles = vector
                 nextNode.position = SCNVector3(0,0,0)
-                
+
             default:
                 print("Wrong translation property key word")
-                
+
             }
-            
             self.stackOfTransformationNodes.append(nextNode)
-            self.currentTransformationNode = nextNode
-                        
-        } else {
             
-            let newTransformationNode: TransitionSCNNode = TransitionSCNNode()
+            if (changeCurrentTransitionNode){
```

Comment:
this area of the code could benefit from comments describing sketch scenarios that would trigger this flow